### DISCUSSION TITLE
简悦的高级账户系统是什么

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#1959] [高级账户]
> 这是一篇介绍简悦账户系统的内容，希望对「不了解」简悦的高级账户系统的用户一些认知。

***

## 数据隐私大于一切

简悦认为，无论是 [标注](http://ksria.com/simpread/docs/#/%E6%A0%87%E6%B3%A8) 还是 [稍后读](http://ksria.com/simpread/docs/#/%E7%A8%8D%E5%90%8E%E8%AF%BB)，都是用户产生的内容，简悦无需也没必要干涉这些数据，用户应该 **享有 100% 对这些数据的所有权**，用户可以将它们放在 NAS / 私有盘等需要的地方，可通过 [此方式自动同步](http://ksria.com/simpread/docs/#/%E8%87%AA%E5%8A%A8%E5%90%8C%E6%AD%A5)。

所以，既然没有用户的数据，所以自然 **无需登录系统**。

## 简悦没有登录系统

> **简悦不收集用户的数据**，所以自然就无需登录系统；细节可以看[《简悦为什么不需要登录系统
》](http://kenshin.wang/blog/#/posts/26)

## 用户标识

每安装一次简悦，会自动给用户分配一个唯一的用户标识，你可以理解为「账户」，同时此账户 **自动为简悦的基础账户**。

## 高级账户

当你付费后，将获得到 License（授权码）与 [当前的基础账户绑定](https://github.com/Kenshin/simpread/discussions/1573)，则当前的基础账户升级为高级账户。

因为没有账户系统，所以 **用户标识（UID）就成为了高级账户唯一凭证**。

## 如何「登录」你的高级账户在新设备上

> 简单的说，**含有高级账户的 UID 即作为登录高级账户的评判标准。**

所以，简悦的高级账户登录有多种方式，如：

1. [配置文件登录](https://github.com/Kenshin/simpread/discussions/1571)（修改为高级账户的 UID ）
2. [授权登录](http://ksria.com/simpread/docs/#/%E6%8E%88%E6%9D%83%E7%99%BB%E5%BD%95)

## 一个例子

> 安装简悦 → 自动分配用户标识 1234 → 绑定 License 到 1234 → 1234 即成为高级账户的标识。

总结：任何人使用 1234 都可以 [授权登录](https://github.com/Kenshin/simpread/discussions/1576) 任意一台设备成为高级账户。

PS：1234 只是个例子，简悦会生成一个唯一的标识。

## 一个更通俗的例子

> 简悦的高级账户机制 **类似 iOS Apple Store 的账户系统**，即：我在账户 A 下买了个软件 P，在任意一台手机上设备上登录 A 账户，即可享有 P 在这个手机上的使用权。

**区别**

iOS Apple Store 使用 **Email/Password** 登录，而简悦使用 **用户标识 「登录」**

## 限制

以防万一，简悦做了物理上的登录设备授权限制，即：**一个高级账户可使用三台设备**。

除此之外，**同一时间，用一高级账户使用不同设备的高级账户功能**，也是不被运行的（会被锁）。（被锁后 [可自行解锁](https://github.com/Kenshin/simpread/discussions/1569)

## 三台设备限制的例子

1234 在 10点 登录了 A 设备；又在 10点20，登录了 B 设备，**这是没问题的**。但简悦发现，A 和 B 设备在 10点30，在 **同时使用标注的样式**（简悦高级账户功能），则 **视为非法使用**。

PS：总不能用户会影分身？

**但下面的方式则是被允许的：**

> 1234 在 10点分别登录了 A 设备的 Chrome 和 Edge，在这两个浏览器同时使用标注的样式（简悦高级账户功能），属于正常行为。

> 而后，1234 又在 10点20 登录了 B 设备的 Chrome 和 Edge，在这两个浏览器同时使用标注的样式（简悦高级账户功能），属于正常行为。（这时 A 设备不在活跃）

> 在之后，1234 又在 10点30 登录了 C 设备，但并没有使用简悦的高级账户功能，也属于正常行为。（即便：B 和 C 设备同时活跃）

即：**同一时间，不同设备只能存在一个活跃的 1234 高级账户账户（功能）**。

## 关联阅读

- [如何找回丢失的 UID （无法恢复高级账户资格）](https://github.com/Kenshin/simpread/discussions/1571)
- [当因遗失 UID 无法登录高级账户账户的解决方案](https://github.com/Kenshin/simpread/discussions/1571)

**Comment by PalmerCharles** [comment]
如果不同设备同时使用不同的两种高级功能呢？

**Reply by Kenshin** [reply]
一样不行的。

只要使用同一时间使用高级账户功能就会被锁。

**Comment by im0qianqian** [comment]
你好，请问不同设备之间判定同时使用的时间差有多久呢？比如我在设备 A 处理好了东西，想转手在设备 B 处理，中间可能差个十几秒，这种情况会被锁么？
另外，是不是假如 UID 被泄露了，便没有办法挽回损失，因为其他人可以随意授权设备并使用。

**Reply by Kenshin** [reply]
> 在设备 A 处理好了东西，想转手在设备 B 处理，中间可能差个十几秒，这种情况会被锁么？

大概率会被锁。正常使用不会出现这种情况，简悦的高级账户已经存在两年多了，还没有人因为这个原因找我...

另外，被锁的前提是：**A 和 B 都使用高级账户功能，且在不同设备同时使用**。如果在不同设备上同时操作 A 的高级账户功能和 B 的基础功能，则没有问题。

> 是不是假如 UID 被泄露了，便没有办法挽回损失，因为其他人可以随意授权设备并使用。

是的，UID 相当你的密钥。（但你可以联系我解决）

**Reply by im0qianqian** [reply]
收到！非常感谢！初次使用，解惑了👍

**Comment by richex-cn** [comment]
![Snipaste_2022-11-07_17-38-30](https://user-images.githubusercontent.com/25103518/200277435-9baf03e3-edee-4046-aec5-438b39c86ef0.png)

这里按照字面意思不太好理解，是不是打错词了。

**Reply by Kenshin** [reply]
不被允许

 🌚

**Comment by richex-cn** [comment]
有几个疑问
1. 本人不太熟悉插件开发，想了解一下区分物理设备是如何实现的？
2. 目前是否有（或者未来考虑增加）同一 IP 的识别？对我来说会有在同一网络下使用多台 PC 的场景，这时[如 @im0qianqian 所说](https://github.com/Kenshin/simpread/discussions/1959#discussioncomment-3862073)，极有可能在几十秒或者几分钟内先后在两台设备使用高级功能。担心因此导致被锁的情况。

对于第二点，因为刚开始用还没遇到过，也许已经足够通过自助解锁手动解决这个问题，但可以当个建议考虑下。

**Reply by Kenshin** [reply]
嗨～

> 问题1

因为涉及到一些验证性的逻辑所以暂时不能回复你。 😂 

> 问题2

暂时不会考虑 IP 段方案，至于 https://github.com/Kenshin/simpread/discussions/1959#discussioncomment-3862073 讨论的内容并不在正常使用范围内。

如果真的想要在多台设备使用的话，建议试试同步助手的 [ SAAS 方案](https://github.com/Kenshin/simpread/discussions/3898) 或 [Docker 方案](https://github.com/Kenshin/simpread/discussions/3898)

PS：如果仍想继续讨论这个话题，直接在这里回复我就好～
